### PR TITLE
Publish wait

### PR DESCRIPTION
**Pull request by SimonRichardson**
See: #11

Most if not all call sites do not wait for Publish to return, they treat
the call as a fire and forget. Unfortunately, the code in place always
ensures that the WaitGroup is done and returns a channel. Placing a
large burden on the callback handlers to behave. This has been proven
not to be the case and we can hold on to the goroutine. This looks like
a leak from the graphs, but in reality, handlers are just not completing
in a timely manner and we're left with back pressure on the subscribers.

## TODO:

 - [x] Bump to v2 as this is a breaking change.

### CODE REVIEW

**Review by manadart — APPROVED** [review]
Already paired on this. Great change.

File: simplehub.go [hunk 1]
```diff
@@ -53,17 +53,15 @@ type SimpleHub struct {
 //
 // The channel return value is closed when all the subscribers have been
 // notified of the event.
```
Comment (jameinel) [inline simplehub.go]: The docstring here is now incorrect. There is no 'channel return value'

File: simplehub.go [hunk 2]
```diff
@@ -130,6 +123,18 @@ func (h *SimpleHub) unsubscribe(id int) {
 	}
 }
 
+// Wait takes the returning function from Publish and returns a channel. The
+// channel is closed once the returning function is done.
+func Wait(done func()) <-chan struct{} {
```
Comment (jameinel) [inline simplehub.go:127]: It seems a little odd to call this pubsub.Wait()
WaitToChannel?
Why return just a func() vs returning the WaitGroup?

I guess you wouldn't want to confuse the caller into thinking they might be responsible for Done, nor wanting them to call wait.Add(1)

**Review by jameinel — APPROVED** [review]
I'm happy with this change, but concerned about changing the 'master' branch to v2.
(and while we're at it, we should move away from the 'master' nomenclature)

File: simplehub.go (hunk 2 above)
Comment (SimonRichardson) [reply simplehub.go:127]: Exactly that. I want to hide WaitGroup, so you can't change something you don't own.

**Pull request merged**